src/main.cpp

```diff
@@ -1420,7 +1420,10 @@ bool CTransaction::ConnectInputs(CTxDB& txdb, MapPrevTx inputs,
                     return DoS(100, error("ConnectInputs() : credit doesn't meet requirement for primenode = %lld while you only have %lld", MINIMUM_FOR_PRIMENODE, GetValueOut()));
                 /* Use time instead of block number because we can better
                  * control when a manditory wallet update is required. */
-                if (nTime >= RESET_PRIMERATES) {
+                if (nTime >= REMOVE_PRIMERATES) {
+                    if (nStakeReward > GetProofOfStakeReward(nCoinAge, 5) - GetMinFee() + MIN_TX_FEE)
+                        return DoS(100, error("ConnectInputs() : %s stake reward exceeded", GetHash().ToString().substr(0,10).c_str()));
+                } else if (nTime >= RESET_PRIMERATES) {
```

Comment:
Contrary to other peoples discussion we can't remove this second check because it's used when resyncing the blockchain so yes let's leave it but being as the first statement is already being evaluated to return an error.

Just a matter of personal preference (not saying it needs to be changed); I would use if instead of else if (it will still be checked in order of top-down) I've had some finicky behavior with else if on other projects. I expect this will work fine but I always feel like sharing random bits like that.